Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1203 @@
+            string ipresult;
+
+            // No preference given, so move on to bind addresses.
+            lock (_intLock)
+            {
+                var nc = _bindAddresses.Exclude(_bindExclusions).Where(p => !p.IsLoopback());
+
+                int count = nc.Count();
+                if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses.Equals(IPAddress.IPv6Any)))
+                {
+                    // Ignore IPAny addresses.
+                    count = 0;
+                }
+
+                if (count != 0)
+                {
+                    // Check to see if any of the bind interfaces are in the same subnet.
+
+                    IEnumerable<IPObject> bindResult;
+                    IPAddress? defaultGateway = null;
+
+                    if (isExternal)
+                    {
+                        // Find all external bind addresses. Store the default gateway, but check to see if there is a better match first.
+                        bindResult = nc.Where(p => !IsInLocalNetwork(p)).OrderBy(p => p.Tag);
+                        defaultGateway = bindResult.FirstOrDefault()?.Address;
+                        bindResult = bindResult.Where(p => p.Contains(sourceAddr)).OrderBy(p => p.Tag);
+                    }
+                    else
+                    {
+                        // Look for the best internal address.
+                        bindResult = nc.Where(p => IsInLocalNetwork(p) && p.Contains(sourceAddr)).OrderBy(p => p.Tag);
+                    }
+
+                    if (bindResult.Any())
+                    {
+                        ipresult = FormatIP6String(bindResult.First().Address);
+                        _logger.LogDebug("{0}: GetBindInterface: Has source, found a match bind interface subnets. {1}", sourceAddr, ipresult);
+                        return ipresult;
+                    }
+
+                    if (isExternal && defaultGateway != null)
+                    {
+                        ipresult = FormatIP6String(defaultGateway);
+                        _logger.LogDebug("{0}: GetBindInterface: Using first user defined external interface. {1}", sourceAddr, ipresult);
+                        return ipresult;
+                    }
+
+                    ipresult = FormatIP6String(nc.First().Address);
+                    _logger.LogDebug("{0}: GetBindInterface: Selected first user defined interface. {1}", sourceAddr, ipresult);
+
+                    if (isExternal)
+                    {
+                        // TODO: remove this after testing.
+                        _logger.LogWarning("{0}: External request received, however, only an internal interface bind found.", sourceAddr);
+                    }
+
+                    return ipresult;
+                }
+
+                if (isExternal)
+                {
+                    // Get the first WAN interface address that isn't a loopback.
+                    var extResult = _interfaceAddresses
```

Comment:
`extResult` is enumerated multiple times

Reply:
altered